optimum/fx/quantization.py

```diff
@@ -0,0 +1,232 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import TYPE_CHECKING, Any, Callable, Dict, List, Optional, Tuple, Type, Union
+
+import torch
+from torch.fx.node import Argument, Node, Target
+from torch.nn.intrinsic import _FusedModule
+from torch.quantization.fx.graph_module import GraphModule, ObservedGraphModule
+from torch.quantization.quantize_fx import Scope, ScopeContextManager
+from torch.quantization.quantize_fx import fuse_fx as orig_fuse_fx
+from torch.quantization.quantize_fx import prepare_fx as orig_prepare_fx
+from torch.quantization.quantize_fx import prepare_qat_fx as orig_prepare_qat_fx
+from transformers.utils.fx import HFTracer, check_if_model_is_supported, get_concrete_args, symbolic_trace
+
+from .utils import check_if_available
+
+
+if TYPE_CHECKING:
+    from torch.fx import Graph
+    from transformers import PreTrainedModel
+
+
+class QuantizationTracer(HFTracer):
+    specialized_concrete_args: Optional[Dict[str, Any]] = None
+
+    def __init__(self, skipped_module_names: List[str], skipped_module_classes: List[Callable]):
```

Comment:
Could we add a docstring here and also for the public methods?

Reply:
Is it needed since I'm just overriding methods that already have a docstring in the base class?

Reply:
Ah sorry I missed that - indeed we can ignore them here in that case